segmentstore/storage/impl/src/test/java/io/pravega/segmentstore/storage/impl/filesystem/IdempotentStorageTestBase.java

```diff
@@ -0,0 +1,245 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.storage.impl.filesystem;
+
+import io.pravega.segmentstore.contracts.BadOffsetException;
+import io.pravega.segmentstore.contracts.StreamSegmentNotExistsException;
+import io.pravega.segmentstore.storage.SegmentHandle;
+import io.pravega.segmentstore.storage.Storage;
+import io.pravega.segmentstore.storage.StorageTestBase;
+import io.pravega.test.common.AssertExtensions;
+import lombok.val;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.ByteArrayInputStream;
+import java.util.concurrent.CompletableFuture;
+
+import static io.pravega.test.common.AssertExtensions.assertMayThrow;
+import static io.pravega.test.common.AssertExtensions.assertThrows;
+
+/**
+ * Unit tests for FileSystemStorage.
+ */
+public abstract class IdempotentStorageTestBase extends StorageTestBase {
+
+
+    //region Fencing tests
+
+    /**
+     * Tests fencing abilities. We create two different Storage objects with different owner ids.
+     * * We create the Segment on Storage1:
+     * ** We verify that Storage1 can execute all operations.
+     * ** We verify that Storage2 can execute only read-only operations.
+     * ** We open the Segment on Storage2:
+     * ** We verify that Storage1 can execute only read-only operations.
+     * ** We verify that Storage2 can execute all operations.
+     */
+    @Override
+    @Test(timeout = 30000)
+    public void testFencing() throws Exception {
+        final long epoch1 = 1;
+        final long epoch2 = 2;
+        final String segmentName = "segment";
+        try (val storage1 = createStorage();
```

Comment:
FYI: If you want to avoid the indentation in cases like this, you can annotate the member with @Cleanup.